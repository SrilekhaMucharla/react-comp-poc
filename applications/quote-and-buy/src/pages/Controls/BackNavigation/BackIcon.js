/* eslint-disable max-len */
import React from 'react';

export default () => (
    <svg width="8" height="14" viewBox="0 0 8 14" xmlns="http://www.w3.org/2000/svg">
        <path d="M0.225564 6.49312C0.0845862 6.6422 -2.91944e-07 6.8211 -2.84124e-07 7C-2.75001e-07 7.20872 0.0845862 7.38761 0.225564 7.50688L5.69549 13.2913C5.83647 13.4404 6.00564 13.5 6.17481 13.5C6.37218 13.5 6.54135 13.4404 6.65414 13.2913L7.27444 12.6353C7.41541 12.5161 7.5 12.3372 7.5 12.1284C7.5 11.9495 7.44361 11.7706 7.30263 11.6216L2.93233 7L7.30263 2.37844C7.44361 2.25917 7.5 2.08028 7.5 1.87156C7.5 1.69266 7.41541 1.51376 7.27444 1.36468L6.65414 0.708715C6.54135 0.58945 6.37218 0.5 6.17481 0.5C6.00564 0.5 5.83647 0.58945 5.69549 0.708715L0.225564 6.49312Z" fill="#0069CC" />
    </svg>
);