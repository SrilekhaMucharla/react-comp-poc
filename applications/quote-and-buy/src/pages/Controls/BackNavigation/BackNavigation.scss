.go-back {
    margin-bottom: 30px;
    &.btn {
        height: auto;
        width: auto;
        font-family: $font-face-generic-regular;
        font-size: $font-size-lg;
        font-style: normal;
        font-weight: 500;
        line-height: 1.35;
        display: flex;
        align-items: center;
        color: $white-color;
        cursor: pointer;
        padding: 4px 13px;
        border: 0 solid transparent !important;
        border-radius: 2px;
        flex-direction: row;
        background: transparent;
        justify-content: flex-start;

        @include media-breakpoint-down(md) {
            padding-left: 4px;
            margin-bottom: 23px;
        }

        .back-icon {
            display: inline-flex;
            margin-right: 8px;
        }

        i {
            margin-right: 12px;
            font-size: 22px;
        }

        &:hover {
            background: $blue-transparent-30;

        }

        &:active {
            outline: 0 transparent;
            background: $mono-big-stone;
        }

        &:focus {
            outline: 0 transparent;
            background: transparent;
        }
        &:disabled {
            background-color: unset;
            svg path {
                fill: $blue-light-2;
                fill-opacity: .3;
            }
        }

        .theme-white &,
        &.theme-white {
            color: $app-theme;

            &:disabled {
                color: $mono-light-steel;
                svg path {
                    fill: $blue-light-6;
                    fill-opacity: .1;
                }
            }
        }
    }
}
