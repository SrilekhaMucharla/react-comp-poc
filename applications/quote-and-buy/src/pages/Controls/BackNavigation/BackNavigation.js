/* eslint-disable react/prop-types */
import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import { HDButtonRefactor } from 'hastings-components';
import * as Message from '../ControlConst';
import BackIcon from './BackIcon';
import { HOMEPAGE } from '../../../constant/const';
import { AnalyticsHDButton as HDButton } from '../../../web-analytics';

const BackNavigation = (props) => {
    const history = useHistory();

    const {
        onKeyPress, onClick, className, disabled
    } = props;

    // Takes user directly to previous screen based on routing history
    // const handleBack = () => history.goBack();
    const handleBack = () => {
        if (props.pathNav) {
            history.push({
                pathname: props.pathNav,
                state: { singleCar: true }
            });
        } else if (history.location && history.location.pathname && history.location.pathname.includes('/intro')) {
            window.location.assign(HOMEPAGE);
        } else {
            history.goBack();
        }
    };

    return (
        // will be replced by back button component
        <HDButton
            webAnalyticsEvent={{ event_action: Message.goback }}
            id="backNavMain"
            name="go-back"
            className={`go-back ${className}`}
            variant="default"
            label={Message.goback}
            onKeyPress={onKeyPress || handleBack}
            onClick={onClick || handleBack}
            disabled={disabled}
        >
            <span className="back-icon"><BackIcon /></span>
        </HDButton>
    );
};

BackNavigation.propTypes = {
    onKeyPress: PropTypes.func,
    onClick: PropTypes.func,
    className: PropTypes.string,
    disabled: PropTypes.bool
};

BackNavigation.defaultProps = {
    onKeyPress: null,
    onClick: null,
    className: '',
    disabled: undefined
};

export default BackNavigation;
