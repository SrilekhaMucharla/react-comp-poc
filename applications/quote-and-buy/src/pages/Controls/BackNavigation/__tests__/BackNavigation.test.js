import React from 'react';
import { shallow, mount } from 'enzyme';
// eslint-disable-next-line no-unused-vars
import { useHistory } from 'react-router-dom';
import BackNavigation from '../BackNavigation';

// given
const mockedGoBack = jest.fn();
jest.mock('react-router-dom', () => ({
    useHistory: () => ({
        goBack: mockedGoBack
    })
}));

describe('BackNavigation', () => {
    it('should render correctly and match the snapshot', () => {
        // given
        const component = shallow(<BackNavigation />);
        // then
        expect(component).toMatchSnapshot();
    });

    it('should navigate back when clicked', () => {
        // given
        const component = mount(<BackNavigation />);
        const button = component.find('[name="go-back"]').at(0);
        // when
        button.simulate('click');
        // then
        expect(mockedGoBack).toHaveBeenCalled();
    });
});
