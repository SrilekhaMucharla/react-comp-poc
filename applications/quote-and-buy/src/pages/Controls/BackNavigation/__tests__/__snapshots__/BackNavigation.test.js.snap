// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackNavigation should render correctly and match the snapshot 1`] = `
<ForwardRef
  className="go-back "
  id="backNavMain"
  label="Go back"
  name="go-back"
  onBeforeOpen={null}
  onBlur={null}
  onCancel={null}
  onChange={null}
  onClick={[Function]}
  onConfirm={null}
  onDelete={null}
  onEdit={null}
  onFocus={null}
  onKeyPress={[Function]}
  onSelect={null}
  variant="default"
  webAnalyticsEvent={
    Object {
      "event_action": "Go back",
    }
  }
>
  <span
    className="back-icon"
  >
    <_default />
  </span>
</ForwardRef>
`;
